hnn_core/dipole.py

```diff
@@ -53,10 +57,17 @@ def simulate_dipole(net, n_trials=None, record_vsoma=False,
 
     # XXX needed in mpi_child.py:run()#L103; include fix in #211 or later PR
     net._params['N_trials'] = n_trials
+
+    # XXX used in network_builder::_simulate_single_trial
+    net._params['tstop'] = tstop
+    net._params['dt'] = dt
+
     net._instantiate_drives(n_trials=n_trials)
     net.cell_response.reset()  # see #290 for context; relevant for MPI
     net._reset_rec_arrays()
 
+    net.cell_response._times = np.arange(0., tstop + dt, dt)
```

Comment:
@rythorpe @cjayb I'd like to get your opinion here. Suppose the user does the following:

```py
net = Network(params)
net.add_xxx_drive(..., tstop=70)
simulate_dipole(..., tstop=40)
```

what's supposed to happen? Currently the drive events are clipped beyond `net.cell_response._times[-1]` which is set from `params['tstop']`. I feel the `Network` object should not know about `tstop`. Would it make sense to pass `tstop` to `_instantiate_drives` explicitly?

Reply:
see: https://github.com/jasmainak/hnn-core/pull/2/files for WIP of the proposal

Reply:
Just to be clear, it should be noted that we're using the same name for two different variables: there is the drive `tstop` that marks the end of a spike event process (whether, repetitive or not), and there is the simulation `tstop` that marks the end of the simulation.

The only reason I can think of why `Network` needs to know the simulation `tstop` is to ensure that whatever drive events are stored in `Network.external_drives` get relayed to `Network.cell_response`. Maybe we're okay with these being different (e.g., if the the user wants to specify a drive with some spike events that fall outside the bounds of the simulation, maybe they should be allowed to). If so, we can just adjust the tests to only test for congruence between those events that are < `tstop`. Otherwise, I think passing the simulation `tstop` to `_instantiate_drives()` is a good solution.

Reply:
I think I'm with Ryan's first option: if a Poisson process is set to stop after the simulation stops, well, let it be so. The `tstop` in `simulate_dipole` will automatically determine the events that get included.

Reply:
So we're okay with the events in `Network.external_drives` and `Network.cell_response` being different? If so, let's at least warn the user in function docstrings.

Reply:
I'm confused. `cell_responses` doesn't contain artificial cell events, does it? In my head the two lists are independent

Reply:
> I'm confused. `cell_responses` doesn't contain artificial cell events, does it? In my head the two lists are independent

I'm pretty sure it does - that's how we plot the input spike histograms. Unless this recently got changed.